### PR TITLE
change subprocess Popen to run

### PR DESCRIPTION
**Pull request by gyorb**
Using subprocess run instead of Popen. It is recommended since python 3.5.

Part of #3020

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3203)

### CODE REVIEW

**Comment by gyorb** [comment]
I've found a `check_output` call here:
https://github.com/qutebrowser/qutebrowser/blob/43aa7423abb931ddcbccc51411f6a8a16bf056b5/tests/unit/utils/test_version.py#L370-L381

Should that be changed to `run` too?

**Comment by The-Compiler** [comment]
Yes - it patches the subprocess call in `version.py` in order to raise an error, so if you change the call in `version.py`, you'll need to adjust the test accordingly for it to pass again.

**Comment by gyorb** [comment]
I've updated it. Should I set the `check=True` attribute on the mock object too or it is not necessary? With the `side_effect` the mock will always throw an exception? I'm not familiar with the mock module. `subprocess.run` throws an exception only if `check` is set to True.

**Comment by The-Compiler** [comment]
It should be fine as-is - the idea is to mock away `subprocess.run` so it *always* raises an exception, to see that `version._git_str_subprocess` handles that case correctly. No need to 100% imitate the same behavior.

**Comment by The-Compiler** [comment]
Sorry I haven't noticed this earlier, but there are some `subprocess.call` calls as well, those should use `run` too. After that, I think we have everything 😆

**Comment by gyorb** [comment]
I've changed the `subprocess.call` too.  Let me know if I missed something.

**Comment by The-Compiler** [comment]
Thank you!

File: qutebrowser/app.py [hunk 1]
```diff
@@ -660,9 +660,9 @@ def restart(self, pages=(), session=None, override_args=None):
         try:
             args, cwd = self._get_restart_args(pages, session, override_args)
             if cwd is None:
-                subprocess.Popen(args)
+                subprocess.run(args)
             else:
-                subprocess.Popen(args, cwd=cwd)
+                subprocess.run(args, cwd=cwd)
```
Comment (The-Compiler) [inline qutebrowser/app.py]: Turns out `subprocess.Popen` was actually the right thing to use here, as we *don't* want to wait for the child process at all here. I fixed this in 4d8ac7486ccdd512b9a4f5d753be79aef418478c.

**Pull request merged**